src/codegen.zig

```diff
@@ -2586,10 +2584,74 @@ fn Function(comptime arch: std.Target.Cpu.Arch) type {
                     },
                     .register => return self.fail(src, "TODO implement genSetReg for aarch64 {}", .{mcv}),
                     .memory => |addr| {
-                        // The value is in memory at a hard-coded address.
```

Comment:
The doc comments for the `memory` tag of the `MCValue` enum says:

```zig
            /// The value is in memory at a hard-coded address.
            /// If the type is a pointer, it means the pointer address is at this memory location.
```

I don't think it makes sense to put the logic here - instead of making it work differently for PIE targets and violating the doc comments for this enum tag, the `MCValue` should be populated with a different tag other than `memory` when doing PIE, *at the location that sets the value*. You can introduce a new `MCValue` tag with a special meaning if you need to.

I'd be ok with making this review comment a follow-up issue.

Reply:
I totally agree with you that this should really be a different tag, I just didn't wanna change too many things at once before we get something working on the Silicon. I'll be more than happy to work on this in a subsequent PR(s), as I'd love to generalise and clean it up!